icu4c/source/test/intltest/caltest.cpp

```diff
@@ -3711,4 +3720,717 @@ void CalendarTest::TestChineseCalendarMapping() {
+        "Europe/Monaco",
+        "Europe/Moscow",
+        "Europe/Nicosia",
+        "Europe/Oslo",
+        "Europe/Paris",
+        "Europe/Podgorica",
+        "Europe/Prague",
+        "Europe/Riga",
+        "Europe/Rome",
+        "Europe/Samara",
+        "Europe/San_Marino",
+        "Europe/Sarajevo",
+        "Europe/Simferopol",
+        "Europe/Skopje",
+        "Europe/Sofia",
+        "Europe/Tallinn",
+        "Europe/Tirane",
+        "Europe/Tiraspol",
+        "Europe/Ulyanovsk",
+        "Europe/Uzhgorod",
+        "Europe/Vaduz",
+        "Europe/Vienna",
+        "Europe/Vilnius",
+        "Europe/Volgograd",
+        "Europe/Warsaw",
+        "Europe/Zagreb",
+        "Europe/Zaporozhye",
+        "Europe/Zurich",
+        "Indian/Chagos",
+        "Indian/Cocos",
+        "Indian/Comoro",
+        "Indian/Kerguelen",
+        "Indian/Mahe",
+        "Indian/Maldives",
+        "Indian/Reunion",
+        "Mexico/BajaSur",
+        "Mexico/General",
+        "Pacific/Apia",
+        "Pacific/Auckland",
+        "Pacific/Bougainville",
+        "Pacific/Chuuk",
+        "Pacific/Enderbury",
+        "Pacific/Fakaofo",
+        "Pacific/Fiji",
+        "Pacific/Galapagos",
+        "Pacific/Gambier",
+        "Pacific/Guadalcanal",
+        "Pacific/Guam",
+        "Pacific/Honolulu",
+        "Pacific/Kosrae",
+        "Pacific/Kwajalein",
+        "Pacific/Majuro",
+        "Pacific/Marquesas",
+        "Pacific/Midway",
+        "Pacific/Nauru",
+        "Pacific/Niue",
+        "Pacific/Norfolk",
+        "Pacific/Noumea",
+        "Pacific/Pago_Pago",
+        "Pacific/Palau",
+        "Pacific/Pohnpei",
+        "Pacific/Ponape",
+        "Pacific/Saipan",
+        "Pacific/Samoa",
+        "Pacific/Tarawa",
+        "Pacific/Truk",
+        "Pacific/Wake",
+        "Pacific/Wallis",
+        "Pacific/Yap",
+        "US/Alaska",
+        "US/Arizona",
+        "US/Hawaii",
+        "US/Michigan",
+        "US/Pacific",
+        "US/Pacific-New",
+        "US/Samoa",
+    };
+    std::string locale;
+    for (int32_t tzidx = 0; tzidx < UPRV_LENGTHOF(tz); tzidx++) {
+        locale = "en-u-tz-";
+        locale += tz[tzidx];
+        Locale l(locale.c_str());
+        UErrorCode status = U_ZERO_ERROR;
+        std::unique_ptr<TimeZoneFormat> tzfmt(TimeZoneFormat::createInstance(l, status));
+        if (U_FAILURE(status)) {
+            dataerrln("Error TimeZoneFormat::createInstance(%s) - %s", locale.c_str(), u_errorName(status));
+            continue;
+        }
+        std::unique_ptr<Calendar> cal(Calendar::createInstance(l, status));
+        if (U_FAILURE(status)) {
+            dataerrln("Error Calendar::createInstance(%s) - %s", locale.c_str(), u_errorName(status));
+            continue;
+        }
+        UnicodeString id;
+        tzfmt->format(UTZFMT_STYLE_ZONE_ID_SHORT, cal->getTimeZone(), now, id);
+        if (id !=  tz[tzidx]) {
+          std::string id_str;
+          id.toUTF8String(id_str);
```

Comment:
Don't assume that the platform uses UTF-8; it could be EBCDIC etc.
Use default-charset conversion instead.
Probably easier to convert the tz string to a UnicodeString and use assertEquals().
In fact, `id !=  tz[tzidx]` under the covers already converts the tz string to a UnicodeString.